edn/tests/tests.rs

```diff
@@ -1003,6 +1003,14 @@ fn test_utils_merge() {
     test(&right, &left, &expected);
 }
 
+macro_rules! def_test_as_value_type {
```

Comment:
Testing with macros feels somewhat dangerous to me, especially if we're implementing with macros. Real world usage is likely to be less hygienic and there's a risk of parallel systemic failures in the impl and test.
That said, this feels simple enough that mistakes are unlikely?